test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c

```diff
@@ -95,12 +95,16 @@ void requestJob_Mqtt_harness()
 {
     OtaAgentContext_t agent;
     OtaInterfaces_t otaInterface;
+    size_t size;
+
+    __CPROVER_assume( size < 128U );
```

Comment:
I'm a little confused by this, we're declaring a size_t variable here and then immediately checking the value is under 128. Where it then gets used as an index a few lines later without being set? Is there something specific to CBMC proofs that is why this is done?

Reply:
This is specific to CBMC proofs. When you declare a variable - size_t for example - CBMC will take on ANY value the data type supports unless you bound it. Line 100 bounds the size variable to be anything less than 128. Then by the time this variable is used, it is initialized by CBMC.